auth/src/main/java/com/vimeo/networking2/Authenticator.kt

```diff
@@ -247,6 +247,51 @@ interface Authenticator {
         callback: VimeoCallback<VimeoAccount>
     ): VimeoRequest
 
+    /**
+     * Find a supported SSO domain that matches the [domain] parameter.
+     *
+     * @param domain A domain, also known as hostname, that might be supported for SSO by the Vimeo API.
+     * @param callback Callback to be notified of the result of the request.
+     *
+     * @return A [VimeoRequest] object to cancel API requests.
+     */
+    @Internal
+    fun fetchSsoDomain(
+        domain: String,
+        callback: VimeoCallback<SsoDomain>
+    ): VimeoRequest
+
+    /**
+     * Obtain the URI which can be used to log in the user and receive back a URI that can be exchanged using
+     * [ssoCodeGrant] for a logged in account. Obtain the [SsoDomain] required by this function first using
+     * [fetchSsoDomain].
+     *
+     * @param ssoDomain The domain info obtained from the API that specifies the [SsoDomain.connectUrl] which will be
+     * used as the base for this URI.
+     * @param responseCode The response code that can be used to identify the origin of the redirect URI.
+     *
+     * @return The URI which can be opened in a browser.
+     */
+    @Internal
+    fun createSsoGrantAuthorizationUri(ssoDomain: SsoDomain, responseCode: String): String
+
+    /**
+     * Authenticate with the server using an authorization code grant from a supported enterprise SSO domain.
+     *
+     * @param uri The URI which was redirected back to you. Must contain a `code` query parameter that contains the
+     * authorization code.
+     * @param marketingOptIn True if the user is opting into marketing emails, false otherwise.
+     * @param callback Callback to be notified of the result of the request.
+     *
+     * @return A [VimeoRequest] object to cancel API requests.
+     */
+    @Internal
+    fun ssoCodeGrant(
```

Comment:
The other (public) code grant function is called `authenticateWithCodeGrant`. I did not call this `authenticateWithSsoCodeGrant` because I was unsure if it would confuse consumers or not, as the API is extremely similar both literally and semantically, but hits different endpoints.

It would be good to unify the naming scheme for authentication functions similar to how `VimeoApiClient` uses `create-`, `edit-`, `delete-`, and `fetch-` prefixes.

I would propose `authorizeWith-` as the designated prefix for all authentication functions. This would result in
- `authenticateWithClientCredentials`
- `authenticateWithGoogle`
- `authenticateWithFacebook`
- `authenticateWithEmailJoin`
- `authenticateWithEmailLogin`
- `exchangeAccessToken` - unchanged
- `exchangeOAuth1Token` - unchanged
- `authenticateWithCodeGrant` - unchanged
- `authenticateWithSsoCodeGrant` - unchanged, this should probably change so that it's not clashing with the other code grant function, but i don't have any ideas
- `authenticateWithPinCode` - unchanged
- `logOut`, `fetch-`, and `create-` functions remain unchanged.

Reply:
I like adding the `authorizeWith` prefix. 

> authenticateWithSsoCodeGrant - unchanged, this should probably change so that it's not clashing with the other code grant function, but i don't have any ideas

Could this be `authenticateWithSso`? I think removing the code grant part would remove some possible confusion or clashing.

Reply:
I think `authenticateWithSso` is a good idea, thanks.